core/unit_test/sycl/TestSYCL_InterOp_Graph.cpp

```diff
@@ -0,0 +1,82 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <filesystem>
+
+#include <TestSYCL_Category.hpp>
+#include <TestGraph_helpers.hpp>
+#include <Kokkos_Graph.hpp>
+
+#include <gtest/gtest.h>
+
+namespace {
+
+// This test checks the promises of Kokkos::Graph against its
+// underlying SYCL native objects.
+TEST(TEST_CATEGORY, graph_promises_on_native_objects) {
+  auto graph = Kokkos::Experimental::create_graph<Kokkos::SYCL>();
+
+  auto root = Kokkos::Impl::GraphAccess::create_root_ref(graph);
+
+  auto graph_ptr_impl =
+      Kokkos::Impl::GraphAccess::get_graph_weak_ptr(root).lock();
+
+  static_assert(std::is_same_v<decltype(graph_ptr_impl)::element_type,
+                               Kokkos::Impl::GraphImpl<Kokkos::SYCL>>);
+
+  // Before instantiation, the SYCL graph is valid, but the SYCL executable
+  // graph is still null. Since the SYCL command graph is a regular object,
+  // no check is needed.
+  // However, the executable SYCL command graph is stored as an optional,
+  // so let's check it is empty for now.
+  ASSERT_FALSE(graph_ptr_impl->get_sycl_graph_exec().has_value());
+
+  // After instantiation, both native objects are valid.
+  graph.instantiate();
+
+  ASSERT_TRUE(graph_ptr_impl->get_sycl_graph_exec().has_value());
+}
+
+// Use native SYCL graph to generate a DOT representation.
+TEST(TEST_CATEGORY, graph_instantiate_and_debug_dot_print) {
+  using view_t = Kokkos::View<int, Kokkos::SYCL>;
+
+  const Kokkos::SYCL exec{};
+
+  view_t data(Kokkos::view_alloc(exec, "witness"));
+
+  auto graph = Kokkos::Experimental::create_graph(exec);
+
+  auto root = Kokkos::Impl::GraphAccess::create_root_ref(graph);
+
+  root.then_parallel_for(1, SetViewToValueFunctor<Kokkos::SYCL, int>{data, 1});
+
+  auto graph_ptr_impl =
+      Kokkos::Impl::GraphAccess::get_graph_weak_ptr(root).lock();
+
+  graph.instantiate();
+
+  ASSERT_EQ(graph_ptr_impl->get_sycl_graph().get_nodes().size(), 2u);
+
+  const auto dot = std::filesystem::temp_directory_path() / "sycl_graph.dot";
+
+  graph_ptr_impl->get_sycl_graph().print_graph(dot, true);
+
+  ASSERT_TRUE(std::filesystem::exists(dot));
+  ASSERT_GT(std::filesystem::file_size(dot), 0u);
```

Comment:
Why didn't you check that the node is in the `dot` file for SYCL?

Reply:
Done. I was able to compile it with my setup but could not execute the test.

Reply:
Did you forget to push the change?

Reply:
Oups. I pushed to my own fork, but this PR is from a branch of our organization :dizzy_face:

Reply:
I check that this works on the testbeds.